cellrank/tl/kernels/_velocity_kernel.py

```diff
@@ -502,15 +521,19 @@ def _run_stochastic(
             # compute the Hessian tensor, and turn it into a matrix that has the diagonal elements in its rows
             W = expression[nbhs_ixs, :] - expression[ix, :]
 
-            H = _predict_transition_probabilities_jax_H(
-                v, W, softmax_scale=softmax_scale
-            )
-            H_diag = np.array([np.diag(h) for h in H])
+            H = scheme.hessian(v, W, softmax_scale)
+            if H.shape == (n_neigh, n_genes):
+                H_diag = H
+            elif H.shape == (n_neigh, n_genes, n_genes):
+                H_diag = np.array([np.diag(h) for h in H])
+            else:
+                raise ValueError(
+                    f"Expected full Hessian matrix of shape `{(n_neigh, n_genes, n_genes)}` "
+                    f"or its diagonal of shape `{(n_neigh, n_genes)}`, found `{H.shape}`."
+                )
 
             # compute zero order term
-            p_0, c = _predict_transition_probabilities_numpy(
-                v[None, :], W, softmax_scale=softmax_scale
-            )
+            p_0, c = scheme(v[None, :], W, softmax_scale)
 
             # compute second order term (note that the first order term cancels)
             p_2 = 0.5 * H_diag.dot(variance[ix])
```

Comment:
One could mention somewhere that our formulas for the expected transition probabilities hold as long as we assume gene-wise velocities to be independent, which is the case in the current scvelo framework. If this ever gets extended, we need to re-work our uncertainty propabation.